monitor.cpp

```diff
@@ -5,34 +5,42 @@
 #include <iostream>
 using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
+void displayMessage(std::string message) {
+    cout << message << "\n";
     for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+        cout << "\r* " << flush;
+        sleep_for(seconds(1));
+        cout << "\r *" << flush;
+        sleep_for(seconds(1));
     }
+}
+
+bool isTemperatureCritical(float temperature) {
+    if (temperature > 102 || temperature < 95) {
+        displayMessage("Temperature is critical!");
     return 0;
-  } else if (pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
     }
-    return 0;
-  } else if (spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+return 1;
+}
+
+
+bool isPulseRateOutOfRange(float pulseRate) {
+    if (pulseRate < 60 || pulseRate > 100) {
```

Comment:
Can further make an abstraction like `isParameterInRange(float inputValue, float lowerLimit, float upperLimit)`
Advantage is that we don't need to add 3 tests for every additional parameter to achieve coverage.